common/libimage/manifests/manifests.go

```diff
@@ -606,22 +610,28 @@ func (l *list) Add(ctx context.Context, sys *types.SystemContext, ref types.Imag
 				instanceInfo.ArtifactType = m.ArtifactType
 			}
 		}
+		instanceInfo.MediaType = primaryManifestType
+		instanceInfo.Size = int64(len(primaryManifestBytes))
 		instanceInfos = append(instanceInfos, instanceInfo)
 	}
 
 	knownConfigTypes := []string{manifest.DockerV2Schema2ConfigMediaType, v1.MediaTypeImageConfig}
 	for _, instanceInfo := range instanceInfos {
+		var hasPlatformConfig bool
 		unparsedInstance := image.UnparsedInstance(src, instanceInfo.instanceDigest)
-		manifestBytes, manifestType, err := unparsedInstance.Manifest(ctx)
-		if err != nil {
-			return "", fmt.Errorf("reading manifest from %q, instance %q: %w", transports.ImageName(ref), instanceInfo.instanceDigest, err)
-		}
-		instanceManifest, err := manifest.FromBlob(manifestBytes, manifestType)
-		if err != nil {
-			return "", fmt.Errorf("parsing manifest from %q, instance %q: %w", transports.ImageName(ref), instanceInfo.instanceDigest, err)
+		if mb, mt, err := unparsedInstance.Manifest(ctx); err == nil {
+			if instanceManifest, err := manifest.FromBlob(mb, mt); err == nil {
+				instanceDigest, err := manifest.Digest(mb)
+				if err != nil {
+					return "", fmt.Errorf("computing digest of manifest from %q: %w", transports.ImageName(ref), err)
+				}
+				instanceInfo.instanceDigest = &instanceDigest
+				instanceInfo.MediaType = mt
+				instanceInfo.Size = int64(len(mb))
+				instanceInfo.ConfigInfo = instanceManifest.ConfigInfo()
+				hasPlatformConfig = instanceInfo.ArtifactType == "" && slices.Contains(knownConfigTypes, instanceInfo.ConfigInfo.MediaType)
```

Comment:
It’s not clear to me why silently setting `hasPlatformConfig` to `false` on an I/O error is OK.